core/ledger/kvledger/txmgmt/statedb/statecouchdb/cache.go

```diff
@@ -64,16 +65,16 @@ func (c *cache) getState(chainID, namespace, key string) (*CacheValue, error) {
 	}
 
 	cacheKey := constructCacheKey(chainID, namespace, key)
-	valBytes := cache.Get(nil, cacheKey)
-	if valBytes == nil {
-		return nil, nil
-	}
 
-	cacheValue := &CacheValue{}
-	if err := proto.Unmarshal(valBytes, cacheValue); err != nil {
-		return nil, err
+	if valBytes, exists := cache.HasGet(nil, cacheKey); exists {
+		cacheValue := &CacheValue{}
+		if err := proto.Unmarshal(valBytes, cacheValue); err != nil {
+			return nil, err
+		}
+		return cacheValue, nil
+
 	}
-	return cacheValue, nil
+	return nil, nil
```

Comment:
nit: better to negate the condition (`!exists` instead of `exists`) and return early.